docs/docs/wikitext103.md

```diff
@@ -0,0 +1,205 @@
+tokenizer = torch.hub.load('huggingface/transformers', 'tokenizer', 'transfo-xl-wt103').to("cuda")
+
+with WikiText103Evaluator.dataset.get_path('local/dir/wikitext-103').open() as f:
+    test_data = torch.tensor(tokenizer.encode(f.read())).to("cuda").eval()
+
+def log_probs_generator(model, test_data, seq_len=128):
+    with torch.no_grad():
+        X, Y, mems = test_data[None, :-1], test_data[None, 1:], None
+        for s in range(0, X.shape[-1], seq_len):
+            log_probs, mems, *_ = model(input_ids=X[s:s+seq_len], mems=mems)
+            yield log_probs, y[s:s+seq_len]
+
+WikiText103Evaluator(
+    model_name="Transformer-XL Large", 
+    paper_arxiv_id="1901.02860",
+    paper_pwc_id="transformer-xl-attentive-language-models" 
+).eval(log_probs_generator(model, test_data)).print_results()
+```
+## Getting Started
+
+You'll need the following in the root of your repository:
+
+- `sotabench.py` file - contains benchmarking logic; the server will run this on each commit
+- `requirements.txt` file - Python dependencies to be installed before running `sotabench.py`
+- `sotabench_setup.sh` *(optional)* - any advanced dependencies or setup, e.g. compilation
+
+You can write whatever you want in your `sotabench.py` file to get langauge model predictions on the WikiText-103 dataset.
+
+But you will need to record your results for the server, and you'll want to avoid doing things like
+downloading the dataset on the server. So you should:
+
+- **Point to the server WikiText-103 data path** - popular datasets are pre-downloaded on the server.
+- **Include an Evaluation object** in `sotabench.py` file to record the results.
+- **Use Caching** *(optional)* - to speed up evaluation by hashing the first batch of predictions.
+
+We explain how to do these various steps below.
+
+## Server Data Location
+
+The WikiText-103 development data is located in the root of your repository on the server at `.data/nlp/wikitext-103/wikitext-103-v1.zip`.
+The zip file has default format of WikiText-103 and contains a folder `wikitext-103` with the following files:
+
+- `wiki.train.tokens`
+- `wiki.valid.tokens`
+- `wiki.test.tokens`
+
+Your local files may have a different file directory structure, so you
+can use control flow like below to change the data path if the script is being
+run on sotabench servers:
+
+``` python
+from sotabencheval.utils import is_server
+
+if is_server():
+    DATA_ROOT = '.data/nlp/squad'
+else: # local settings
+    DATA_ROOT = '/home/ubuntu/my_data/'
+```
+This will detect if `sotabench.py` is being run on the server and change behaviour accordingly.
+
+Alternatively you can use our helper function that will automatically unpack the zip for you and point you to the right location as follows:
+
+```python
+from sotabencheval.langauge_modelling import WikiText103Evaluator
+
+test_file_path = WikiText103Evaluator.dataset.get_path('/home/ubuntu/data')
+```
```

Comment:
Unzipping automatically is fine, but not sure about this get_path method. in any case, can we pass DATA_ROOT into this? also it's confusing because I don't know what I'm supposed to do with this?